contracts/interchain-token-service/src/contract/execute/interceptors.rs

```diff
@@ -205,13 +205,15 @@ fn destination_token_decimals(
         state::load_chain_config(storage, destination_chain).change_context(Error::State)?;
 
     if source_chain_config
+        .truncation
         .max_uint
-        .le(&destination_chain_config.max_uint)
+        .le(&destination_chain_config.truncation.max_uint)
     {
         source_chain_decimals
     } else {
-        source_chain_config
-            .max_target_decimals
+        destination_chain_config
+            .truncation
+            .max_decimals_when_truncating
             .min(source_chain_decimals)
```

Comment:
Can you rename this to 
```suggestion
            .min(source_token_decimals)
```

Reply:
I like that renaming